Core/src/main/java/org/tribuo/ONNXExportable.java

```diff
@@ -47,9 +58,39 @@
      */
     public static final String PROVENANCE_METADATA_FIELD = "TRIBUO_PROVENANCE";
 
+    /**
+     * Creates an ONNX model protobuf for the supplied context.
+     *
+     * @param onnxContext  The context which contains the ONNX graph.
+     * @param domain       Domain for the produced model.
+     * @param modelVersion Model version for the produced model.
+     * @param model        Provenanced Tribuo model from which this model is derived - the DocString and Tribuo Provenance data
+     *                     from this model will be written into the ONNX Model proto.
+     * @param <M>          The type of the provenanced model.
+     * @return An ONNX model proto of the graph represented by the supplied ONNXContext.
+     */
+    public static <M extends Provenancable<ModelProvenance>> OnnxMl.ModelProto buildModel(ONNXContext onnxContext, String domain, long modelVersion, M model) {
+        return OnnxMl.ModelProto.newBuilder()
+                .setGraph(onnxContext.buildGraph())
+                .setDomain(domain)
+                .setProducerName("Tribuo")
+                .setProducerVersion(Tribuo.VERSION)
+                .setModelVersion(modelVersion)
+                .addOpsetImport(ONNXOperators.getOpsetProto())
+                .setIrVersion(6)
+                .setDocString(model.toString())
+                .addMetadataProps(OnnxMl.StringStringEntryProto
+                        .newBuilder()
+                        .setKey(PROVENANCE_METADATA_FIELD)
+                        .setValue(SERIALIZER.marshalAndSerialize(model.getProvenance()))
+                        .build())
+                .build();
+    }
+
     /**
      * Exports this {@link org.tribuo.Model} as an ONNX protobuf.
-     * @param domain A reverse-DNS name to namespace the model (e.g., org.tribuo.classification.sgd.linear).
+     *
+     * @param domain       A reverse-DNS name to namespace the model (e.g., org.tribuo.classification.sgd.linear).
      * @param modelVersion A version number for this model.
      * @return The ONNX ModelProto representing this Tribuo Model.
      */
```

Comment:
Given the new `buildModel` method above, shouldn't this be deprecated or removed?

Reply:
Unfortunately there's no typing relationship between `ONNXExportable` and `Model` so I can't get access to the members necessary to default the method, and I don't really want to make `ONNXExportable` have a bunch of extra methods to expose that information as it makes adding the interface to things like `SequenceModel` tricky.